ai-ml/trainium-inferentia/helm-values/karpenter-resources/values.yaml

```diff
@@ -0,0 +1,52 @@
+# Shared values
+name: default
+clusterName:
+
+# EC2NodeClass specific values
+ec2NodeClass:
+  enabled: true
+  amiFamily: AL2
+  amiSelectorTerms:
+  subnetSelectorTags:
+    kubernetes.io/role/internal-elb: "1"  # Added here to make sure that will be scheduled in private subnets
+  karpenterRole:
+  metadataOptions:
+    httpEndpoint: enabled
+    httpProtocolIPv6: disabled
+    httpPutResponseHopLimit: 2
+    httpTokens: required
+  blockDevice:
+    deviceName: /dev/xvda
+    volumeSize: 100Gi
+    volumeType: gp3
+    encrypted: true
+    deleteOnTermination: true
+  detailedMonitoring: true
+  userData:
+
+# NodePool specific values
+nodePool:
+  enabled: true
+  labels:
+  taints:
+  requirements:
+    - key: "karpenter.k8s.aws/instance-family"
+      operator: In
+      values: ["c5", "m5", "r5"]
+    - key: "karpenter.k8s.aws/instance-size"
+      operator: In
+      values: ["xlarge", "2xlarge", "4xlarge", "8xlarge", "16xlarge", "24xlarge"]
+    - key: "kubernetes.io/arch"
+      operator: In
+      values: ["amd64"]
+    - key: "karpenter.sh/capacity-type"
+      operator: In
+      values: ["spot", "on-demand"]
```

Comment:
These configurations should be explicitly set in the user-defined values instead of being predefined as defaults within the Helm Chart.